doc/decisions/backend_orm_crate.md

```diff
@@ -0,0 +1,70 @@
+# Backend ORM/SQL crate
+
+## Problem
+
+We are currently using [diesel](https://github.com/diesel-rs/diesel) as our ORM in Rust.  
+Since diesel doesn't support async out of the box we will look at other alteratives and see if a switch is reasonable.  
+There might also be advantages of other ORM crates that as of writing this have not been considered.
+
+## Constraints
+
+1. Migrations need to be supported
+
+## Assumptions
+
+1. diesel is a solid choice, therefore unless other ORMs have obvious advantages differences will not be looked at in detail
+
+## Solutions
+
+### [diesel](https://github.com/diesel-rs/diesel)
+
+Diesel currently is the most used ORM in the Rust ecosystem.  
+It supports migrations written in SQL that can be executed from command line.
+
+Diesel is the ORM we are using as of writing this decisions.
+
+### [diesel](https://github.com/diesel-rs/diesel) + [diesel_async](https://github.com/weiznich/diesel_async)
+
+diesel_async is a drop-in replacement of corresponding diesel methods providing async support.
+
+### [sea-orm](https://github.com/SeaQL/sea-orm)
+
+Sea-orm is based upon [sqlx](https://github.com/jmoiron/sqlx) and supports async out of the box.  
+It supports [migrations](https://www.sea-ql.org/SeaORM/docs/migration/setting-up-migration/) written in Rust.
+
```

Comment:
An alternative solution is that we review for correct async usage and you help if needed.

Reply:
In my opinion that is not really an alternative solution, but would rather mean we just stick with diesel (without diesel_async). So I'm not adding a separate alternative.

Apart from that I'm not a fan of that as it would require manual checking every time and can be quite tricky to spot. For example currently simply calling any function in the `service` directory would lead to blocking of the executor. You cannot really spot blocking functions without actually knowing the whole codebase and which are blocking.
The same applies for locks and the likes. That is also why there are async lock such as [async RwLock](https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html).

I added that to the decision as well.